assets/js/base/components/product-list/container.js

```diff
@@ -1,47 +1,45 @@
 /**
  * External dependencies
  */
-import { Component } from 'react';
+import { useState } from '@wordpress/element';
 import PropTypes from 'prop-types';
-import withQueryStringValues from '@woocommerce/base-hocs/with-query-string-values';
 
 /**
  * Internal dependencies
  */
 import ProductList from './index';
 
-class ProductListContainer extends Component {
-	onPageChange = ( newPage ) => {
-		this.props.updateQueryStringValues( {
-			product_page: newPage,
-		} );
+const ProductListContainer = ( {
+	attributes,
+	// eslint-disable-next-line camelcase
+	product_page = 1,
+	// eslint-disable-next-line camelcase
+	product_sort,
+} ) => {
+	const [ currentPage, setPage ] = useState( product_page );
+	const [ currentSort, setSort ] = useState(
+		// eslint-disable-next-line camelcase
+		product_sort || attributes.orderby
+	);
+	const onPageChange = ( newPage ) => {
+		setPage( newPage );
 	};
-
-	onSortChange = ( event ) => {
+	const onSortChange = ( event ) => {
 		const newSortValue = event.target.value;
-		this.props.updateQueryStringValues( {
-			product_sort: newSortValue,
-			product_page: 1,
-		} );
+		setSort( newSortValue );
+		setPage( 1 );
 	};
 
-	render() {
-		// eslint-disable-next-line camelcase
-		const { attributes, product_page, product_sort } = this.props;
-		const currentPage = parseInt( product_page );
-		const sortValue = product_sort || attributes.orderby; // eslint-disable-line camelcase
-
-		return (
-			<ProductList
-				attributes={ attributes }
-				currentPage={ currentPage }
-				onPageChange={ this.onPageChange }
-				onSortChange={ this.onSortChange }
-				sortValue={ sortValue }
-			/>
-		);
-	}
-}
+	return (
+		<ProductList
+			attributes={ attributes }
+			currentPage={ currentPage }
+			onPageChange={ onPageChange }
+			onSortChange={ onSortChange }
+			sortValue={ currentSort }
+		/>
+	);
+};
 
 ProductListContainer.propTypes = {
 	attributes: PropTypes.object.isRequired,
```

Comment:
In the previous comment I suggested removing the `product_page` and `product_sort` props, but if we don't do this, we still need to remove the comment in the line below this one: `// From withQueryStringValues`.